### PR TITLE
Improvements to navigation

### PR DESCRIPTION
**Pull request by yourordinarycat**
**Resolved / Related Issues**
I don't know if there's any on GitHub, but it is a known problem we've been trying to fix for a while.

**Details of Changes**
We used to save the state of NavigationView items during app suspension. However, this approach was somewhat problematic:
- The suspending event has a deadline - a pretty short one at that
- During suspension, we get low priority access to resources, which means we can't always save state fast enough
- If the app crashes, everything's up in the air, we can't expect suspension to just work

Navigation state is now saved when any modification that requires serializing again takes place. There's also a lot of improvements to the overall structure of the code, and we can add or remove items across app updates. Performance wise, deserialization should be way faster - on my end I noticed up to a 1s improvement in some cases.

All playlists are now pinned to the NavigationView as well.

**Validation**
How did you test these changes?
- [x] Built and ran the app in debug and release mode - we had some problems with .NET Native in the past after all

**Screenshots / Videos (optional)**
None - the UI didn't really change, aside from playlist pinning options being removed

### CODE REVIEW

**Review by itsWindows11 — APPROVED** [review]
LGTM. Can you add source generation for System.Text.Json? Might improve performance a bit more, but it's fast already.

File: Rise Media Player Dev/Settings/NavigationPage.xaml [hunk 1]
```diff
@@ -6,18 +6,53 @@
...
+            <selectors:NavigationItemTemplateSelector.DestinationTemplate>
+                <DataTemplate x:DataType="nav:NavigationItemDestination">
+                    <labs:SettingsCard Header="{x:Bind markup:ResourceHelper.GetString(Label)}">
+                        <ToggleSwitch
+                            IsOn="{x:Bind IsVisible, Mode=OneWay}"
+                            Loaded="ItemToggleSwitch_Loaded"
+                            Tag="{x:Bind}" />
+
+                        <labs:SettingsCard.HeaderIcon>
+                            <FontIcon Glyph="{x:Bind DefaultIcon}" />
```
Comment (itsWindows11) [inline Rise Media Player Dev/Settings/NavigationPage.xaml:25]: You should use the [WCT's FontIconExtension](https://learn.microsoft.com/en-us/windows/communitytoolkit/extensions/iconmarkupextensions#fonticonextension) instead, as other settings controls use it.

File: Rise Media Player Dev/Settings/NavigationPage.xaml [hunk 2]
```diff
@@ -6,18 +6,53 @@
...
+            <selectors:NavigationItemTemplateSelector.HeaderTemplate>
+                <DataTemplate x:DataType="nav:NavigationItemHeader">
+                    <labs:SettingsCard Header="{markup:ResourceHelper Name=/Settings/NavigationShowHeader}" IsEnabled="{x:Bind IsGroupVisible, Mode=OneWay}">
+                        <ToggleSwitch
+                            IsOn="{x:Bind IsVisible, Mode=OneWay}"
+                            Loaded="ItemToggleSwitch_Loaded"
+                            Tag="{x:Bind}" />
+
+                        <labs:SettingsCard.HeaderIcon>
+                            <FontIcon Glyph="&#xE8D2;" />
```
Comment (itsWindows11) [inline Rise Media Player Dev/Settings/NavigationPage.xaml]: Same as above

File: Rise.Data/Navigation/Items/NavigationItemDestination.cs [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+        /// <summary>
+        /// The item's text content.
+        /// </summary>
+        public string Label { get; init; }
+
+        /// <summary>
+        /// An access key for this item.
+        /// </summary>
+        public string AccessKey { get; init; }
```
Comment (itsWindows11) [inline Rise.Data/Navigation/Items/NavigationItemDestination.cs:26]: Do access keys ever work?

**Review by itsWindows11 — CHANGES_REQUESTED** [review]
Looks mostly good to me, just did a funny for some reason but anyways there are a few changes that should be done.

Also, can you add source generation for navigation in System.Text.Json? Might improve performance a bit more, but it's fast already.

File: Rise Media Player Dev/Settings/NavigationPage.xaml (hunk 1 above)
Comment (yourordinarycat) [reply Rise Media Player Dev/Settings/NavigationPage.xaml:25]: Markup extensions can't be used with Binding or x:Bind, fixed it for headers though.

File: Rise.Data/Navigation/Items/NavigationItemDestination.cs (hunk 3 above)
Comment (yourordinarycat) [reply Rise.Data/Navigation/Items/NavigationItemDestination.cs:26]: They work on my end - they're invoked with Alt + Key

File: Rise Media Player Dev/Settings/NavigationPage.xaml (hunk 1 above)
Comment (itsWindows11) [reply Rise Media Player Dev/Settings/NavigationPage.xaml:25]: Oh right, apparently I was looking at the wrong line.

**Review by itsWindows11 — APPROVED** [review]

**Pull request merged**